### ISSUE TITLE
Empty docs break sort_by()

### ISSUE THREAD

**Issue by matthiasvegh** [mikefarah/yq#2164]
**Describe the bug**

If there are empty documents passed into eval-all, sorting only works before and after the empty document.

Version of yq: 4.43.1
Operating system: Ubuntu 20.04
Installed via: wget https://github.com/mikefarah/yq/releases/download/v4.43.1/yq_linux_amd64

**Input Yaml**
data1.yml:
```yaml
---
kind: Service
metadata:
  name: b
---
kind: Deployment
metadata:
  name: a
---
# Empty document, meaningful
---
kind: ConfigMap
metadata:
  name: e
---
kind: ConfigMap
metadata:
  name: d
``` 

**Command**
The command you ran:
```
yq eval-all '[.] | sort_by(.kind, .metadata.name) | .[] | split_doc'
```

**Actual behavior**

```yaml
kind: Deployment
metadata:
  name: a
---
kind: Service
metadata:
  name: b
---

---
kind: ConfigMap
metadata:
  name: d
---
kind: ConfigMap
metadata:
  name: e
```

**Expected behavior**

```yaml
---
kind: ConfigMap
metadata:
  name: d
---
kind: ConfigMap
metadata:
  name: e
---
kind: Deployment
metadata:
  name: a
---
kind: Service
metadata:
  name: b
```

**Additional context**
It appears that empty documents behave like an ordering boundary, where everything before it is sorted correctly, as is everything after, but not across. Curiously, if instead of `sort_by(.kind, .metadata.name)` I do `sort_by("\(.kind), \(.metadata.name)")` everything is sorted correctly.

My actual motivation is to compare Kubernetes manifests as in #1101. In my case, the manifests are produced by Helm, which can sometimes emit empty documents if the manifest is not needed.

**Comment by krokofant** [comment]
This became very confusing in our custom diffing workflow when we accidentally left extra `---` somewhere in the yaml 😅

Would be nice with a cleaner workaround that more expressively _filter out empty documents_ before doing the sort.

**Comment by mikefarah** [comment]
Thanks for raising, will fix in the next release

**Comment by mikefarah** [comment]
Fixed in 4.44.5
